include/boost/math/special_functions/jacobi_theta.hpp

```diff
@@ -0,0 +1,830 @@
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta2tau(T z, U tau);
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta3tau(T z, U tau);
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta4tau(T z, U tau);
+
+// Minus one versions for small q / large tau
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta3m1(T z, U q);
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta4m1(T z, U q);
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta3m1tau(T z, U tau);
+template <class T, class U>
+inline typename tools::promote_args<T, U>::type jacobi_theta4m1tau(T z, U tau);
+
+// Policied versions - parameterized by q
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta1(T z, U q, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta2(T z, U q, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta3(T z, U q, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta4(T z, U q, const Policy& pol);
+
+// Policied versions - parameterized by tau
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta1tau(T z, U tau, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta2tau(T z, U tau, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta3tau(T z, U tau, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta4tau(T z, U tau, const Policy& pol);
+
+// Policied m1 functions
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta3m1(T z, U q, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta4m1(T z, U q, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta3m1tau(T z, U tau, const Policy& pol);
+template <class T, class U, class Policy>
+inline typename tools::promote_args<T, U>::type jacobi_theta4m1tau(T z, U tau, const Policy& pol);
+
+// Internal convergence criterion:
+// If the delta is more than an absolute epsilon, keep going.
+// If the delta is more than an relative epsilon, and the denominator
+// is non-zero, keep going.
+template <class RealType>
+inline bool
+_jacobi_theta_converged(RealType result, RealType delta, RealType eps) {
+    return abs(delta) < eps && (abs(result) == 0.0 || abs(delta/result) < eps);
```

Comment:
You need a using std::abs here.

Reply:
I will plan to eliminate `abs` when I revamp this function per other discussions.